src/modules/item/sagas.spec.ts

```diff
@@ -0,0 +1,197 @@
+import * as matchers from 'redux-saga-test-plan/matchers'
+import { BuilderAPI } from '../../lib/api/builder'
+import { saveItemFailure, saveItemRequest, saveItemSuccess, savePublishedItemRequest, savePublishedItemSuccess } from './actions'
+import { itemSaga } from './sagas'
+import { Item, ItemType } from './types'
+import { calculateFinalSize } from './export'
+import { MAX_FILE_SIZE } from './utils'
+import { call, select } from 'redux-saga/effects'
+import { Collection } from 'modules/collection/types'
+import { getItems } from './selectors'
+import { getCollections } from 'modules/collection/selectors'
+import { ChainId, WearableBodyShape, WearableCategory } from '@dcl/schemas'
+import { getChainIdByNetwork } from 'decentraland-dapps/dist/lib/eth'
+import { sendTransaction } from 'decentraland-dapps/dist/modules/wallet/utils'
+
+describe('Item sagas', () => {
+  let blob: Blob = new Blob()
+  const contents: Record<string, Blob> = { path: blob }
+
+  const builderAPI = ({
+    saveItem: jest.fn(),
+    saveItemContents: jest.fn()
+  } as unknown) as BuilderAPI
+
+  let dateNowSpy: jest.SpyInstance
+  const updatedAt = Date.now()
+
+  beforeEach(() => {
+    dateNowSpy = jest.spyOn(Date, 'now').mockImplementation(() => updatedAt)
+  })
+
+  afterEach(() => {
+    dateNowSpy.mockRestore()
+  })
+
+  describe('when handling the save item request action', () => {
+    it('should throw if name contains ":"', () => {
+      const item = {
+        name: 'invalid:name'
+      } as Item
+      return expectSaga(itemSaga, builderAPI)
+        .put(saveItemFailure(item, contents, 'Invalid character! The ":" is not allowed in names or descriptions'))
+        .dispatch(saveItemRequest(item, contents))
+        .run({ silenceTimeout: true })
+    })
+
+    it('should throw if description contains ":"', () => {
+      const item = {
+        name: 'valid name',
+        description: 'invalid:description'
+      } as Item
+      return expectSaga(itemSaga, builderAPI)
+        .put(saveItemFailure(item, contents, 'Invalid character! The ":" is not allowed in names or descriptions'))
+        .dispatch(saveItemRequest(item, contents))
+        .run({ silenceTimeout: true })
+    })
+
+    it('should throw if description contains ":"', () => {
+      const item = {
+        name: 'valid name',
+        description: 'invalid:description'
+      } as Item
+      return expectSaga(itemSaga, builderAPI)
+        .put(saveItemFailure(item, contents, 'Invalid character! The ":" is not allowed in names or descriptions'))
+        .dispatch(saveItemRequest(item, contents))
+        .run({ silenceTimeout: true })
+    })
+
+    it('should throw if file size is larger than 2 MB', () => {
+      const item = {
+        name: 'valid name',
+        description: 'valid description'
+      } as Item
+      return expectSaga(itemSaga, builderAPI)
+        .provide([[matchers.call.fn(calculateFinalSize), Promise.resolve(MAX_FILE_SIZE + 1)]])
```

Comment:
You could also use `.provide([[call(calculateFinalSize, expectedItem, expectedContent), Promise.resolve(MAX_FILE_SIZE + 1)]])` To also validate that the function is receiving the expected parameters